### PR TITLE
feat: create or update nodeinfo crd at cns startup for multitenancy

### PR DESCRIPTION
**Pull request by thatmattlong**
**Reason for Change**:
CNS will write the vm unique id to the nodeinfo crd (or update it, if one already exists) for the node at startup if swift v2 is enabled. This is required because we can't reliably read the vm unique id from the node.Status.NodeInfo.SystemUniqueID property, as it can be reported by kubelet/cadvisor with the wrong endianess.

**Requirements**:

- [x] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [x] adds unit tests
- [x] relevant PR labels added

**Notes**:

### CODE REVIEW

File: cns/service/main.go [hunk 1]
```diff
@@ -1215,8 +1219,9 @@ func InitializeCRDState(ctx context.Context, httpRestService cns.HTTPService, cn
 	if _, ok := node.Labels[configuration.LabelNodeSwiftV2]; ok {
 		cnsconfig.EnableSwiftV2 = true
 		cnsconfig.WatchPods = true
-		// TODO(rbtr): create the NodeInfo for Swift V2
-		// register the noop mtpnc reconciler to populate the cache
+		if nodInfoErr := createOrUpdateNodeInfoCRD(ctx, kubeConfig, node); nodInfoErr != nil {
```
Comment (aggarwal0009) [inline cns/service/main.go:1222]: nit; nodeInfo

**Review by aggarwal0009 — APPROVED** [review]
LGTM

File: crd/multitenancy/client.go [hunk 2]
```diff
@@ -190,3 +191,28 @@ func (i *Installer) InstallOrUpdatePodNetworkInstance(ctx context.Context) (*v1.
 	}
 	return current, nil
 }
+
+type NodeInfoClient struct {
```
Comment (nddq) [inline crd/multitenancy/client.go:195]: Looks like we have implemented an `Installer` struct which seems abstracted enough to be used for NodeInfo CRD as well, is there a reason why we can't build upon that?

Comment (rbtr) [reply crd/multitenancy/client.go:195]: `Installer` is for the CustomResource**Definition**, not for the instance of the CustomResource

Comment (nddq) [reply crd/multitenancy/client.go:195]: ah i see what you mean, not familiar with CNS creating the actual CR itself 🙂

**Review by miguelgoms — DISMISSED** [review]

File: cns/service/main.go (hunk 1 above)
Comment (thatmattlong) [reply cns/service/main.go]: fixed

**Review by aggarwal0009 — APPROVED** [review]

**Review by nddq — APPROVED** [review]

**Review by miguelgoms — APPROVED** [review]

**Pull request merged**